### ISSUE TITLE
Cargo.toml version, features & dependency suggestion

### ISSUE THREAD

**Issue by JustFrederik** [zed-industries/zed#10630]
### Check for existing issues

- [X] Completed

### Describe the feature

code completion when writing dependency names/versions/features

### If applicable, add mockups / screenshots to help present your vision of the feature

https://github.com/Saecki/crates.nvim

**Comment by JosephTLyons** [comment]
Not necessarily the same request, but linking these, as they go hand in hand:

- https://github.com/zed-industries/zed/issues/7081

**Comment by JustFrederik** [comment]
I played around with a language server that uses the crates.io http api. I tossed this tho, because the api takes quite long and has bad results. I can't accept 1 second response times. When searching for 'ran' 'rand' isn't even on the first page. Im going to retry it with the [crates.io-index](https://github.com/rust-lang/crates.io-index). This is quite big tho(2.2G). It would be optimal if someone would host a service that strips the unimportant data like dependencies from that file. It can be shrunk down to 14mb(only name & versions) or 74mb(name&versions&features)
